src/rgw/rgw_op.cc

```diff
@@ -4948,12 +4948,12 @@ bool RGWCopyObj::parse_copy_location(const std::string_view& url_src,
   if (dec_src[0] == '/')
     dec_src.remove_prefix(1);
 
-  pos = dec_src.find('/');
+  pos = url_decode(dec_src).find('/');
   if (pos == string::npos)
     return false;
 
-  bucket_name = url_decode(dec_src.substr(0, pos));
-  key.name = url_decode(dec_src.substr(pos + 1));
+  bucket_name = url_decode(dec_src).substr(0, pos);
+  key.name = url_decode(dec_src).substr(pos + 1);
```

Comment:
instead of calling `url_decode()` 3 times, please store the result in a temporary string

we have a variable `std::string_view dec_src`, but it doesn't represent the decoded form, so i'd rather use that variable name for this new temporary string

so i'd suggest this change instead:
```diff
-  std::string_view dec_src{name_str};
-  if (dec_src[0] == '/')
-    dec_src.remove_prefix(1);
+  if (name_str[0] == '/') // trim leading slash
+    name_str.remove_prefix(1);
+
+  std::string dec_src = url_decode(name_str);
```
what do you think?

Reply:
Makes sense to me, thanks! Made those changes now